### PR TITLE
`Modeling exercises`: Modeling exercises implementation

### PR DESCRIPTION
**Pull request by AlexanderG2207**
~~⚠️ [PR#32](https://github.com/ls1intum/artemis-ios-core-modules/pull/32) in the Core Modules should be merged first, and the Core Modules dependency rule in this PR should be changed back before merging ⚠️~~ 

This PR adds support and allows users to participate in modeling exercises within the Artemis iOS app.

Screenshot 2023-11-29 at 14 57 55

### CODE REVIEW

File: ArtemisKit/Sources/CourseView/ExerciseTab/ExerciseDetailView.swift [hunk 1]
```diff
@@ -67,13 +69,46 @@ public struct ExerciseDetailView: View {
         }
     }
 
+    private var isExerciseParticipatable: Bool {
```
Comment (nityanandaz) [inline ArtemisKit/Sources/CourseView/ExerciseTab/ExerciseDetailView.swift:72]:
```suggestion
    private var isExerciseParticipationAvailable: Bool {
```

File: ArtemisKit/Sources/CourseView/ExerciseTab/ExerciseDetailView.swift [hunk 2]
```diff
@@ -250,10 +280,81 @@ private struct ExerciseDetailCell<Content: View>: View {
     }
 }
 
-private struct FeedbackView: View {
+private struct StartExerciseButton: View {
```
Comment (nityanandaz) [inline ArtemisKit/Sources/CourseView/ExerciseTab/ExerciseDetailView.swift:283]: Let's think about adding the destinations to the NavigationController

File: feature/CourseView/Sources/CourseView/ExerciseTab/ExerciseParticipation/ModelingExerciseParticipation/ModelingExerciseView.swift [hunk 3]
```diff
@@ -9,6 +9,7 @@ import ApollonEdit
 import DesignLibrary
 import SharedModels
 import SwiftUI
+import ApollonShared
 
 struct ModelingExerciseView: View {
```
Comment (nityanandaz) [inline feature/CourseView/Sources/CourseView/ExerciseTab/ExerciseParticipation/ModelingExerciseParticipation/ModelingExerciseView.swift:14]: Sources should not be located in `feature/CourseView/Sources`, but in `ArtemisKit`

- ModelingExerciseView
- ModelingViewModel
- ExerciseSubmissionService
- ModelingExerciseSubmissionServiceImpl
- UnknownExerciseSubmissionServiceImpl

File: ArtemisKit/Sources/CourseView/ExerciseParticipation/ModelingExercise/Views/EditModelingExerciseView.swift [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+                                isGridBackground: true)
+                }
+            } else {
+                ArtemisHintBox(text: R.string.localizable.diagramTypeNotSupported(), hintType: .warning)
+                    .padding(.horizontal, .l)
+            }
+        }
+        .task {
+            await modelingVM.onAppear()
+            modelingVM.setup()
```
[the remark this answers is not in the document]
Comment (AlexanderG2207) [reply ArtemisKit/Sources/CourseView/ExerciseParticipation/ModelingExercise/Views/EditModelingExerciseView.swift]: I gave both functions more descriptive names. They should be kept separate.

**Review by nityanandaz — APPROVED** [review]

**Review by nityanandaz — APPROVED** [review]

**Pull request merged**